lib/ansible/modules/network/iosxr/iosxr_netconf.py

```diff
@@ -0,0 +1,200 @@
+        if have['state'] == 'absent':
+            commands.append('netconf-yang agent ssh')
+
+        if want['netconf_port'] != have.get('netconf_port'):
+            commands.append(
+                'ssh server netconf port %s' % want['netconf_port']
+            )
+        if want['netconf_vrf'] != have.get('netconf_vrf'):
+            commands.append(
+                'ssh server netconf vrf %s' % want['netconf_vrf']
+            )
+
+    return commands
+
+
+def parse_vrf(config):
+    match = re.search(r'vrf (\w+)', config)
+    if match:
+        return match.group(1)
+
+
+def parse_port(config):
+    match = re.search(r'port (\d+)', config)
+    if match:
+        return int(match.group(1))
+
+
+def map_config_to_obj(module):
+    obj = {'state': 'absent'}
+
+    netconf_config = get_config(module, flags=['netconf-yang agent'])
+
+    ssh_config = get_config(module, flags=['ssh server'])
+    ssh_config = [config_line for config_line in (line.strip() for line in ssh_config.splitlines()) if config_line]
+
+    for config in ssh_config:
+        if 'netconf port' in config:
+            obj.update({'netconf_port': parse_port(config)})
+        if 'netconf vrf' in config:
+            obj.update({'netconf_vrf': parse_vrf(config)})
+    if 'ssh' in netconf_config or 'netconf_port' in obj or 'netconf_vrf' in obj:
+        obj.update({'state': 'present'})
+
+    if 'ssh' in netconf_config and 'netconf_port' not in obj:
+        obj.update({'netconf_port': 830})
+    return obj
+
+
+def validate_netconf_port(value, module):
+    if not 1 <= value <= 65535:
+        module.fail_json(msg='netconf_port must be between 1 and 65535')
+
+
+def map_params_to_obj(module):
+    obj = {
+        'netconf_port': module.params['netconf_port'],
+        'netconf_vrf': module.params['netconf_vrf'],
+        'state': module.params['state']
+    }
+
+    for key, value in iteritems(obj):
+        # validate the param value (if validator func exists)
+        validator = globals().get('validate_%s' % key)
+        if callable(validator):
+            validator(value, module)
+
+    return obj
+
+
+def main():
+    """main entry point for module execution
+    """
+    argument_spec = dict(
+        netconf_port=dict(type='int', default=830, aliases=['listens_on']),
+        netconf_vrf=dict(aliases=['vrf']),
+        state=dict(default='present', choices=['present', 'absent']),
+    )
+    argument_spec.update(iosxr_argument_spec)
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True)
+
+    warnings = list()
+    check_args(module, warnings)
+
+    result = {'changed': False, 'warnings': warnings}
+
+    want = map_params_to_obj(module)
+    have = map_config_to_obj(module)
+    commands = map_obj_to_commands((want, have), module)
+    result['commands'] = commands
+
+    if commands:
+        if not module.check_mode:
+            load_config(module, commands, result['warnings'], commit=True)
```

Comment:
Load config returns diff of the configuration. Can the diff be added as part of result if `--diff` option is enabled.
```
        diff = load_config(module, commands, result['warnings'], commit=True)
        if diff:
            if module._diff:
                result['diff'] = {'prepared': diff}
```